File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOHeartRateModeParamModel.swift
//
//  IDOHeartRateModeParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set Heart Rate Mode Event
@objcMembers
public class IDOHeartRateModeParamModel: NSObject, IDOBaseModel {
    /// 0: Off
    /// 1: Auto(5min)
    /// 2: Continuous Monitoring(5sec)
    /// 3: Manual Mode
    public var mode: Int
    /// Time Range
    /// 0: No
    /// 1: Yes
    public var hasTimeRange: Int
    /// Start Hour (24-hour format, 0-23)
    public var startHour: Int
    /// Start Minute (0-59)
    public var startMinute: Int
    /// End Hour
    public var endHour: Int
    /// End Minute
    public var endMinute: Int
    /// Measurement Interval
    /// Unit: minute
    public var measurementInterval: Int
    
    enum CodingKeys: String, CodingKey {
        case mode = "mode"
        case hasTimeRange = "has_time_range"
        case startHour = "start_hour"
        case startMinute = "start_minute"
        case endHour = "end_hour"
        case endMinute = "end_minute"
        case measurementInterval = "measurement_interval"
    }
    
    public init(mode: Int,hasTimeRange: Int,startHour: Int,startMinute: Int,endHour: Int,endMinute: Int,measurementInterval: Int) {
        self.mode = mode
        self.hasTimeRange = hasTimeRange
        self.startHour = startHour
        self.startMinute = startMinute
        self.endHour = endHour
        self.endMinute = endMinute
        self.measurementInterval = measurementInterval
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOHeartRateModeParamModel.kt
//
//  IDOHeartRateModeParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set Heart Rate Mode Event

open class IDOHeartRateModeParamModel(
    mode: Int,
    hasTimeRange: Int,
    startHour: Int,
    startMinute: Int,
    endHour: Int,
    endMinute: Int,
    measurementInterval: Int
) : IDOBaseModel {

    /// 0: Off
    /// 1: Auto(5min)
    /// 2: Continuous Monitoring(5sec)
    /// 3: Manual Mode
    @SerializedName("mode")
    var mode: Int = mode

    /// Time Range
    /// 0: No
    /// 1: Yes
    @SerializedName("has_time_range")
    var hasTimeRange: Int = hasTimeRange

    /// Start Hour (24-hour format, 0-23)
    @SerializedName("start_hour")
    var startHour: Int = startHour

    /// Start Minute (0-59)
    @SerializedName("start_minute")
    var startMinute: Int = startMinute

    /// End Hour
    @SerializedName("end_hour")
    var endHour: Int = endHour

    /// End Minute
    @SerializedName("end_minute")
    var endMinute: Int = endMinute

    /// Measurement Interval
    /// Unit: minute
    @SerializedName("measurement_interval")
    var measurementInterval: Int = measurementInterval


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    